### ISSUE TITLE
Provide a Semigroupal instance for SortedSet

### ISSUE THREAD

**Issue by LukaJCB** [typelevel/cats#2708]
Should be something like this:

```scala
def product[A, B](fa: SortedSet[A], fb: SortedSet[B]): SortedSet[(A,B)] = {
  implicit val orderingA = fa.ordering
  implicit val orderingB = fb.ordering
    
  fa.flatMap(a => fb.map(b => a -> b))
}
```

**Comment by calvinbrown085** [comment]
Hi @LukaJCB I'd be happy to give this a shot, would that go in here? https://github.com/typelevel/cats/blob/774fb51d1365e1adef7fa71f09b1410941264f60/core/src/main/scala/cats/instances/sortedSet.scala#L9

**Comment by LukaJCB** [comment]
Unfortunately I think it'd need to go in another trait for binary compatability.

**Comment by calvinbrown085** [comment]
@LukaJCB if have a second, do you mind pointing me in the right direction? Or by `another trait` do you mean a new file? That we can create for bin compat issues

**Comment by LukaJCB** [comment]
Something like this one: 
https://github.com/typelevel/cats/blob/774fb51d1365e1adef7fa71f09b1410941264f60/core/src/main/scala/cats/instances/option.scala#L130

You can't extend the trait directly so we mix it into the objects that need it, i.e. `cats.implicits` and `cats.instances.sortedSet`. :)

**Comment by calvinbrown085** [comment]
Perfect, that's what I figured :)

**Comment by calvinbrown085** [comment]
So I have that PR, I'm a bit confused on why it's failing the bin compat check

**Comment by vladislavGutov** [comment]
Hi @LukaJCB 
Seems like this one was forgotten.
I created PR https://github.com/typelevel/cats/pull/2840. When you have time, could look at it, please?

**Comment by calvinbrown085** [comment]
@vladislavGutov Thank you! I haven't had time to look at this unfortunately. I'll close this PR.

**Comment by kailuowang** [comment]
closed by #2840
